### PR TITLE
Resque scheduler

### PR DESCRIPTION
**Pull request by crigor**
Uses Joe Heth's code from https://github.com/engineyard/ey-cookbooks-stable-v5/pull/75

### CODE REVIEW

File: examples/resque_scheduler/cookbooks/custom-resque_scheduler/recipes/default.rb [hunk 1]
```diff
@@ -0,0 +1 @@
+include_recipe 'resque'
```
Comment (dvalfre) [inline examples/resque_scheduler/cookbooks/custom-resque_scheduler/recipes/default.rb]: it should be `include_recipe 'resque_scheduler'`

**Comment by dvalfre** [comment]
The main 'resque-scheduler' recipe has the following code:

```
  execute 'install resque gem' do
    command 'gem install resque redis redis-namespace yajl-ruby -r'
    not_if { 'gem list | grep resque' }
end
```

Which comes from the v4 recipe.  According to https://github.com/engineyard/ey-cloud-recipes/issues/150 that codes is faulty and evaluates to '1', which is why the gems get never installed.
Funnily enough, the 'resque-scheduler' works (in v4, and guess will do in v5) since the necessary Gems are included on the app's Gemfile.

Shall the aforemenationed code be dropped from the main 'resque-scheduler' recipe?

**Comment by radamanthus** [comment]
@dvalfre I agree, the gem dependencies should be in the Gemfile and the recipe should not bother with installing the gems.

File: examples/resque_scheduler/cookbooks/custom-redis/attributes/default.rb [hunk 2]
```diff
@@ -0,0 +1 @@
+default['redis']['utility_name'] = 'resque'
```
Comment (dvalfre) [inline examples/resque_scheduler/cookbooks/custom-redis/attributes/default.rb]: For a *clustered* env,  this line should be replaced with at least:

```
default['redis'].tap do |redis|

  # Collect the redis instances in this array
  redis_instances = []

  # Run Redis on a named util instance
  # This is the default
  redis['utility_name'] = 'resque'
  redis_instances << redis['utility_name']
  redis['is_redis_instance'] = (
    node['dna']['instance_role'] == 'util' &&
    redis_instances.include?(node['dna']['name'])
  )
end
```

and for a *solo* env with:

```
default['redis'].tap do |redis|

  # Collect the redis instances in this array
  redis_instances = []

 # Run redis on a solo instance
  # Not recommended for production environments
  #redis['is_redis_instance'] = (node['dna']['instance_role'] == 'solo') 
end
```

File: examples/resque_scheduler/cookbooks/custom-redis/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Custom Redis
+
+This custom-redis recipe is used in the resque example to install redis on the utility instance named "resque" instead of the default "redis".
```
Comment (dvalfre) [inline examples/resque_scheduler/cookbooks/custom-redis/README.md]: Shall we include this small Redis wrapper at all? Leaning towards adding a section on the Resque-Scheduler wrapper's Readme telling the customer to enable the full Redis recipe (as we do with Sidekiq).

Comment (radamanthus) [inline examples/resque_scheduler/cookbooks/custom-redis/README.md]: I opened PR #150 to remove custom-redis from resque_scheduler

**Pull request merged**